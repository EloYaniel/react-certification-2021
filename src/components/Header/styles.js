import styled from 'styled-components';

export const Header = styled.header`
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-direction: row;
  flex-wrap: wrap;
  background-color: #1c5476;
  width: 100%;
  min-height: 8vh;
  z-index: 1;
  color: #fff;
`;

export const RightContent = styled.div`
  display: flex;
  justify-content: right;
  padding-right: 10px;
  align-items: center;

  @media only screen and (max-width: 600px) {
    display: none;
  }
`;

export const LeftContent = styled.div`
  display: flex;
  justify-content: left;
  width: 50%;
  padding-left: 10px;
`;
