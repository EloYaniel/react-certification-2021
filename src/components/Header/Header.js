import React from 'react';
import { Login } from './Login';
import { Menu } from './Menu';
import { SearchBar } from './SearchBar';
import { ThemeToggle } from './ThemeToggle';
import { Header as StyledHeader, RightContent, LeftContent } from './styles';

export const Header = () => (
  <StyledHeader>
    <LeftContent>
      <Menu />
      <SearchBar />
    </LeftContent>
    <RightContent>
      <ThemeToggle />
      <Login />
    </RightContent>
  </StyledHeader>
);
