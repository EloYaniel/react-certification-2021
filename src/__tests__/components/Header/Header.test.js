import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../../../components/Header';

describe('Header Component tests suit', () => {
  it('Should match snapshot', () => {
    const element = render(<Header />);
    expect(element).toMatchSnapshot();
  });
});
