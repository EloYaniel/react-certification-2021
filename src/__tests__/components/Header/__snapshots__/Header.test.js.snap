// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component tests suit Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-iqHYGH ihpwry"
      >
        <div
          class="sc-dQppl klJPtn"
        >
          <div
            class="sc-dlfnbm cbbQCX"
          >
            <div
              class="sc-hKgILt bzWINk"
            />
            <div
              class="sc-hKgILt bzWINk"
            />
            <div
              class="sc-hKgILt bzWINk"
            />
          </div>
          <input
            class="sc-iBPRYJ gxYavs"
            placeholder="Wizeline"
          />
        </div>
        <div
          class="sc-crrsfI TzpMg"
        >
          <label
            class="sc-fubCfw fKqnzi"
          >
            <input
              class="sc-pFZIQ bOVXbQ"
              type="checkbox"
              value="false"
            />
            <div
              class="sc-jrAGrp ycoes"
            />
            <span
              class="sc-kEjbxe goTjtW"
            >
              Dark mode
            </span>
          </label>
          <a
            class="sc-bdfBwQ jeXWtc"
            href="/login"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-iqHYGH ihpwry"
    >
      <div
        class="sc-dQppl klJPtn"
      >
        <div
          class="sc-dlfnbm cbbQCX"
        >
          <div
            class="sc-hKgILt bzWINk"
          />
          <div
            class="sc-hKgILt bzWINk"
          />
          <div
            class="sc-hKgILt bzWINk"
          />
        </div>
        <input
          class="sc-iBPRYJ gxYavs"
          placeholder="Wizeline"
        />
      </div>
      <div
        class="sc-crrsfI TzpMg"
      >
        <label
          class="sc-fubCfw fKqnzi"
        >
          <input
            class="sc-pFZIQ bOVXbQ"
            type="checkbox"
            value="false"
          />
          <div
            class="sc-jrAGrp ycoes"
          />
          <span
            class="sc-kEjbxe goTjtW"
          >
            Dark mode
          </span>
        </label>
        <a
          class="sc-bdfBwQ jeXWtc"
          href="/login"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
